container/go/pkg/compat/ociToLegacy.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// This utility works with the new_container_pull and new_container_load targets
+// to generate the appropriate pseudo-intermediate format that is compatible
+// with the rules_docker container_import rule.
+
+package compat
+
+import (
+	ospkg "os"
+	"path"
+	"strconv"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/pkg/errors"
+)
+
+// Extension for layers and config files that are made symlinks
+const targzExt = ".tar.gz"
+const configExt = "config.json"
+
+// generateSymlinks safely generates a symbolic link from src to dst.
+func generateSymlinks(src, dst string) error {
+	if _, err := ospkg.Stat(dst); err != nil {
+		return errors.Wrapf(err, "layer file does not exist at %s", dst)
+	}
+
+	if _, err := ospkg.Lstat(src); err == nil {
+		if err = ospkg.Remove(src); err != nil {
+			return errors.Wrapf(err, "failed to remove the file at %s", src)
+		}
+	}
+	if err := ospkg.Symlink(dst, src); err != nil {
+		return errors.Wrapf(err, "failed to create symbolic link from %s to %s", src, dst)
+	}
+
+	return nil
+}
+
+// LegacyFromOCIImage creates predictable symbolic links to the config.json and layer .tar.gz files
+// so that they may be easily consumed by container_import targets.
+// The dstPath is the top level directory in which the puller will create symlinks inside an image/ directory
+// pointing to actual pulled OCI image artifacts in image-oci/ directory.
+func LegacyFromOCIImage(img v1.Image, ociPath, symlinkDir string) error {
```

Comment:
call the arguments `srcDir` and `dstDir`. The name `symlinkDir` exposes an implementation detail and the fact that its going from OCI -> Legacy is apparent from the function name.